internal/provider/data_source_gateway_domain.go

```diff
@@ -42,7 +48,11 @@ func dataSourceGatewayDomain() *schema.Resource {
 // TODO: make this non failing
 func dataSourceGatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	apiClient := meta.(*apiClient)
-	nodeID := uint32(d.Get("node").(int))
+	capacityID := uint32(d.Get("capacity_id").(int))
```

Comment:
i find it strange we convert to uint32 and in the next line we convert to uint64?